temba/api/support.py

```diff
@@ -38,7 +39,34 @@ def authenticate_credentials(self, key):
 
             return token.user, token
 
-        raise exceptions.AuthenticationFailed("User inactive or deleted")
+        raise exceptions.AuthenticationFailed("Invalid token")
+
+
+class APIBasicAuthentication(BasicAuthentication):
+    """
+    Basic authentication.
+
+    Clients should authenticate using HTTP Basic Authentication.
+
+    Credentials: username:api_token
+    """
+
+    def authenticate_credentials(self, userid, password):
+        try:
+            token = APIToken.objects.get(is_active=True, key=password)
+        except APIToken.DoesNotExist:
+            raise exceptions.AuthenticationFailed("Invalid token")
+
+        if token.user.username != userid:
+            raise exceptions.AuthenticationFailed("Invalid username")
```

Comment:
I think this still leak some info, may be error message can be "Authentication failed" in all places

Reply:
I agree, we should log the actual reason why authentication failed, and return generic message to the API consumer

Reply:
Personally, I would prefer to give the API consumer an informative error message so that they can easily tell where they made a mistake. And also since the Token is random and very hard to memorize, I think it would make sense for the user know if the authentication mistake is a result of providing the wrong Token or a result of providing a wrong email address.
I am wondering if it makes sense to instead change both error messages to `Invalid Email or Token` instead of the more generic `Authentication failed`?

Reply:
Probably a more descriptive message like `Authentication failed, please check the email and token` would be enough
The main thing is to keep the same error message for all the failures and not leak information whether the user was deleted or the token expired